Source/Meadow.Foundation.Peripherals/Sensors.Motion.Bmi270/Driver/Bmi270.cs

```diff
@@ -0,0 +1,288 @@
+
+                /* Store 4 to 11 bits of address in the second byte */
+                dmaLocation[1] = (byte)((index / 2) >> 4);
+
+                Thread.Sleep(1); //probably not needed ... data sheet wants a 2us delay
+
+                i2cPeripheral.WriteRegister(INIT_0, dmaLocation);
+
+                i2cPeripheral.WriteRegister(INIT_DATA, bmi270_config_file.Skip(index).Take(length).ToArray());
+
+                index += length;
+            }
+
+            //Write INIT_CTRL 0x01 to complete config load
+            i2cPeripheral.WriteRegister(INIT_CTRL, 1);
+
+            //wait until register INTERNAL_STATUS contains 0b0001 (~20 ms)
+            while (true)
+            {
+                Thread.Sleep(10);
+                byte status = i2cPeripheral.ReadRegister(INTERNAL_STATUS);
+
+                if (status == 0x01) { break; }
+            }
+            //Afer initialization - power mode is set to "configuration mode"
+            //Need to change power modes before you can sample data
+        }
+
+        /// <summary>
+        /// Set the range of values the sensor can read for acceleration
+        /// </summary>
+        /// <param name="accelRange">AccelerationRange</param>
+        public void SetAccelerationRange(AccelerationRange accelRange)
+        {
+            i2cPeripheral.WriteRegister(ACC_RANGE, (byte)accelRange);
+            CurrentAccelerationRange = accelRange;
+        }
+
+        /// <summary>
+        /// Set the range of values the sensor can read for angular velocity (gyro)
+        /// </summary>
+        /// <param name="angRange">AngularAccelerationRange</param>
+        public void SetAngularAccelerationRange(AngularVelocityRange angRange)
+        {   //This register also sets the OIS range but it's not implemented so we can ignore it 
+            i2cPeripheral.WriteRegister(GYR_RANGE, (byte)angRange);
+        }
+
+        protected override void RaiseEventsAndNotify(IChangeResult<(Acceleration3D? Acceleration3D, AngularVelocity3D? AngularVelocity3D, Units.Temperature? Temperature)> changeResult)
+        {
+            if (changeResult.New.AngularVelocity3D is { } angular)
+            {
+                AngularVelocity3DUpdated?.Invoke(this, new ChangeResult<AngularVelocity3D>(angular, changeResult.Old?.AngularVelocity3D));
+            }
+            if (changeResult.New.Acceleration3D is { } accel)
+            {
+                Acceleration3DUpdated?.Invoke(this, new ChangeResult<Acceleration3D>(accel, changeResult.Old?.Acceleration3D));
+            }
+            if (changeResult.New.Temperature is { } temp)
+            {
+                TemperatureUpdated?.Invoke(this, new ChangeResult<Units.Temperature>(temp, changeResult.Old?.Temperature));
+            }
+            base.RaiseEventsAndNotify(changeResult);
+        }
+
+        protected override Task<(Acceleration3D? Acceleration3D, AngularVelocity3D? AngularVelocity3D, Units.Temperature? Temperature)> ReadSensor()
```

Comment:
Couldnt we just make this async and remove the Task.Run wrapper?

Reply:
Unfortunantely no because none of the calls within the method can be awaited